src/datatypes/src/error.rs

```diff
@@ -122,6 +122,13 @@ pub enum Error {
 
     #[snafu(display("Failed to unpack value to given type: {}", reason))]
     TryFromValue { reason: String, location: Location },
+
+    #[snafu(display("Invalid arguments, reason: {}", error))]
+    InvalidArguments {
```

Comment:
question: Why does the invalid arguments error use arrow's error as the source error? We should use another variant name. And we don't print source error message in display.
```suggestion
    #[snafu(display("Invalid arguments"))]
    InvalidArguments {
```